src/main/java/org/jabref/cli/ArgumentProcessor.java

```diff
@@ -239,6 +253,114 @@ public boolean hasParserResults() {
         return loaded;
     }
 
+    private void writeXMPtoPdf(List<ParserResult> loaded, String filesAndCitekeys, JabRefPreferences preferences) {
+
+        ParserResult pr = loaded.get(loaded.size() - 1);
+        BibDatabaseContext databaseContext = pr.getDatabaseContext();
+        BibDatabase dataBase = pr.getDatabase();
+
+        XmpPdfExporter xmpPdfExporter = new XmpPdfExporter(preferences.getXmpPreferences());
+
+        if (filesAndCitekeys.equals("all")) {
+            for (BibEntry entry : dataBase.getEntries()) {
+                writeXMPtoPDFsOfEntry(databaseContext, dataBase, entry, preferences, xmpPdfExporter);
+            }
+            return;
+        }
+
+        Vector<String> citeKeys = new Vector<>();
+        Vector<String> pdfs = new Vector<>();
+        for (String fileOrCiteKey : filesAndCitekeys.split(",")) {
+            if (fileOrCiteKey.toLowerCase(Locale.ROOT).endsWith(".pdf")) {
+                pdfs.add(fileOrCiteKey);
+            } else {
+                citeKeys.add(fileOrCiteKey);
+            }
+        }
+
+        writeXMPtoPdfByCitekey(databaseContext, dataBase, citeKeys, preferences, xmpPdfExporter);
+        writeXMPtoPdfByFileNames(databaseContext, dataBase, pdfs, preferences, xmpPdfExporter);
+
+    }
+
+    private void writeXMPtoPDFsOfEntry(BibDatabaseContext databaseContext, BibDatabase dataBase, BibEntry entry, JabRefPreferences preferences, XmpPdfExporter xmpPdfExporter) {
+        boolean writtenToAFile = false;
+
+        for (LinkedFile file : entry.getFiles()) {
+            if (file.getFileType().toLowerCase(Locale.ROOT).equals("pdf")) {
+                Optional<Path> filePath = file.findIn(databaseContext, preferences.getFilePreferences());
+                if (filePath.isEmpty()) {
+                    System.err.println(Localization.lang("Skipped - PDF %0 does not exist", filePath.get()));
+                } else {
+                    try {
+                        xmpPdfExporter.export(databaseContext, filePath.get(), preferences.getDefaultEncoding(), Arrays.asList(entry));
+                        writtenToAFile = true;
+                    } catch (Exception e) {
+                        System.err.println(Localization.lang("Error writing XMP content: %0", filePath.get()));
+                    }
+                }
+            }
+        }
+
+        if (!writtenToAFile) {
+            System.err.println(Localization.lang("Skipped - No PDF linked to entry %0", entry.getCitationKey()));
+        }
+    }
+
+    private void writeXMPtoPdfByCitekey(BibDatabaseContext databaseContext, BibDatabase dataBase, Vector<String> citeKeys, JabRefPreferences preferences, XmpPdfExporter xmpPdfExporter) {
+        for (String citeKey : citeKeys) {
+            Optional<BibEntry> entry = dataBase.getEntryByCitationKey(citeKey);
```

Comment:
You are using

```
dataBase.getEntryByCitationKey()
```

would

```
dataBase.getEntriesByCitationKey()
```

be more "complete"?

(citation keys should be unique but aren't always)

Reply:
Yay! one more loop! :)